### PR TITLE
refactor(app/module): minor readability improvements and more tests

### PR DESCRIPTION
**Pull request by rootulp**
I was reviewing the `app/module` directory because I want to write a README for it but I noticed a few small refactors that I started working on. The refactors include: 
- adding more unit tests
- simplify how `firstVersion` and `lastVersion` is set in the manager
- extracting types to a `types.go` file
- extracting utils to a `utils.go` file and adding tests
- improving code comments
- renaming `VersionedConfigurator` => `Configurator`
- ordering public methods above private methods
- renaming files to match the type they define (i.e. `manager.go`, `configurator.go`)

### CODE REVIEW

**Review by evan-forbes — APPROVED** [review]

**Review by ninabarbakadze — COMMENTED** [review]
let's resolve lint and i'll take a look.

p.s sorry i merged main. it's a habit

File: app/module/utils.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
+package module
+
+import (
+	"slices"
+	"sort"
+
+	sdkmodule "github.com/cosmos/cosmos-sdk/types/module"
+)
+
+const authName = "auth"
```
Comment (cmwaters) [inline app/module/utils.go:10]: nit: I wonder if you could just get that name from the auth module. Not that it should ever change

**Review by cmwaters — DISMISSED** [review]
Interesting pattern to separate out into types and utils files. I'm not sure if I'd agree that this improves readability but I don't want to bike shed on this

File: app/module/utils.go (hunk 1 above)
Comment (rootulp) [reply app/module/utils.go]: Great call b363d9f3dfdeb1d390a052c9374986bbf517ad3e

**Review by evan-forbes — APPROVED** [review]

**Review by ninabarbakadze — APPROVED** [review]

**Pull request merged**